katex.js

```diff
@@ -47,6 +50,9 @@ if (typeof document !== "undefined") {
  * Parse and build an expression, and return the markup for that.
  */
 var renderToString = function(expression, options) {
+    if (!(typeof expression === 'string' || expression instanceof String)) {
+        throw new TypeError('renderToString requires string typed expression');
+    }
     var settings = new Settings(options);
 
     var tree = parseTree(expression, settings);
```

Comment:
Putting both these checks in `parseTree` would avoid duplication. That way you couldn't include an API-level function name, but usually browsers will indicate `katex.js` as the source of this exception, and for those who need more detail the stack traces will likely be more useful anyway.